prover/src/lib.rs

```diff
@@ -150,6 +148,11 @@ pub trait Prover {
     where
         E: FieldElement<BaseField = Self::BaseField>;
 
+    /// Trace low-degree extension for building the LDEs of trace segments and their commitments.
+    type ConstraintCommitment<E>: ConstraintCommitment<E, HashFn = Self::HashFn, VC = Self::VC>
+    where
+        E: FieldElement<BaseField = Self::BaseField>;
```

Comment:
The comment here needs to be updated (I think it is just copy over from the `TraceLde` comment above).